BrainPortal/lib/scir_lsf.rb

```diff
@@ -28,73 +28,62 @@ class ScirLsf < Scir
 
   class Session < Scir::Session #:nodoc:
 
-    def update_job_info_cache #:nodoc:
-      # FIXME: modify this method based
+    def update_job_info_cache #:nodoc:      
       # on the implementation of bstat.  This method is supposed to
       # list all the statuses of the running tasks, and to keep it in
       # the @job_info_cache array.
-      out, err = bash_this_and_capture_out_err("bstat -f -u #{CBRAIN::Rails_UserName.to_s.bash_escape}")
-      raise "Cannot get output of 'bstat -f' ?!?" if out.blank? && ! err.blank?
+      out, err = bash_this_and_capture_out_err("bjobs -a -noheader -u #{CBRAIN::Rails_UserName.to_s.bash_escape}")
+      raise "Cannot get output of 'bjobs -a' ?!?" if out.blank? && ! err.blank?
       jid = 'Dummy'
       @job_info_cache = {}
       out.split(/\s*\n\s*/).each do |line|
-        line.force_encoding('ASCII-8BIT')  # some pbs 'qstat' commands output junk binary data!
-        if line =~ /^Job\s+id\s*:\s*(\S+)/i
-          jid = Regexp.last_match[1]
-          if jid =~ /^(\d+)/
-            jid = Regexp.last_match[1]
-          end
-          next
-        end
-        next unless line =~ /^\s*job_state\s*=\s*(\S+)/i
-        state = statestring_to_stateconst(Regexp.last_match[1])
-        @job_info_cache[jid.to_s] = { :drmaa_state => state }
+        bjob_stat = line.gsub(/\s+/m, ' ').strip.split(" ")
+        jid = bjob_stat[0]
+        stat = bjob_stat[2]
```

Comment:
I added this line
I have this as output: in the variable @job_info_cache

`ext-ruby-2.2.3-p173 :102 > print @job_info_cache
{"846007"=>{:drmaa_state=>"STATE_DONE"}, "846264"=>{:drmaa_state=>"STATE_DONE"}, "846276"=>{:drmaa_state=>"STATE_DONE"}} => nil
ext-ruby-2.2.3-p173 :103 >`

However, the state in cbrain portal does not get refreshed accordingly.